### PR TITLE
HRCPP-196 The build script should have an option to build the test

### PR DESCRIPTION
**Pull request by alanfx**
executables and link with prebuilt libraries

Added option SKIP_HOTROD_LIBRARY which does not build the dynamic or
static libraries and looks for the prebuilt libraries in the build
directory. By default, this option is off.

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 237](http://ci.infinispan.org/viewLog.html?buildId=27772&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 130fdc8f4b6c61040b1b7ee42ebdf78aa1652de8
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:42

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 223](http://ci.infinispan.org/viewLog.html?buildId=27771&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 130fdc8f4b6c61040b1b7ee42ebdf78aa1652de8
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:45

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 238](http://ci.infinispan.org/viewLog.html?buildId=27773&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 130fdc8f4b6c61040b1b7ee42ebdf78aa1652de8
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:16

File: CMakeLists.txt [hunk 1]
```diff
@@ -120,6 +120,26 @@ endif(HOTROD_WINAPI)
...
+    if("${HOTROD_STATIC_LIBRARY}" STREQUAL "HOTROD_STATIC_LIBRARY-NOTFOUND")
+        message(FATAL_ERROR "Cannot find HotRod static library in your environment.")
+    else("${HOTROD_STATIC_LIBRARY}" STREQUAL "HOTROD_STATIC_LIBRARY-NOTFOUND")
+        add_library(hotrod-static STATIC IMPORTED GLOBAL)
+        set_target_properties(hotrod-static PROPERTIES IMPORTED_LOCATION ${HOTROD_STATIC_LIBRARY})
+    endif("${HOTROD_STATIC_LIBRARY}" STREQUAL "HOTROD_STATIC_LIBRARY-NOTFOUND")
+
+else(SKIP_HOTROD_LIBRARY)
+
 set (library_sources
```
Comment (isavin) [inline CMakeLists.txt]: This block needs indenting.

**Comment by isavin** [comment]
Looks OK to me.

The only comment I have is that it might be a bit more clean to have the library outside the ${CMAKE_CURRENT_BINARY_DIR} which is build_{windows,linux} and to point to it using a cmdline flag -DUSE_PREBUILD_LIB and then instread on redefining "hotrod" and "hotrod-static" define two news ones "hotrod-prebuilt" and "hotrod-prebuilt-static" and based on the cmdline flag use one or the other when linking with the tests (target_link_libraries (itest ${hotrod-test-lib})).

I think it should work but I'm not sure. Let me know if you want to give it a try otherwise I can pull it as it is (+indentation fix).

**Comment by alanfx** [comment]
Hey @isavin I thought about providing the directory to search instead of using CMAKE_CURRENT_BINARY_DIR and change the name to HOTROD_PREBUILT_LIB_DIR. I'm not sure I see the benefit of defining "hotrod-prebuilt" and "hotrod-prebuilt-static". I'll update the PR to make the first change unless you can convince me about the second one! :-)

**Comment by alanfx** [comment]
@isavin the PR has been updated

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 239](http://ci.infinispan.org/viewLog.html?buildId=27805&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 2630c114d346671ccf1269ab2894f5e002c735c6
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:01

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 240](http://ci.infinispan.org/viewLog.html?buildId=27806&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 2630c114d346671ccf1269ab2894f5e002c735c6
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:11

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 224](http://ci.infinispan.org/viewLog.html?buildId=27804&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 2630c114d346671ccf1269ab2894f5e002c735c6
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:40

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 242](http://ci.infinispan.org/viewLog.html?buildId=27810&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:24

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 241](http://ci.infinispan.org/viewLog.html?buildId=27809&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:59

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 225](http://ci.infinispan.org/viewLog.html?buildId=27808&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:49

**Comment by isavin** [comment]
Separate "*-prebuilt-*" libs would have avoided changing the meaning of hotrod and hotrod-static and avoided the if/else loop which redefines their meaning but it's not that important.

Pushed the changes. Thanks!

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 244](http://ci.infinispan.org/viewLog.html?buildId=27820&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:52

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 245](http://ci.infinispan.org/viewLog.html?buildId=27821&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:37

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 227](http://ci.infinispan.org/viewLog.html?buildId=27818&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for de5c8b70b729ee24a27fce62b6e0a7dd83096996
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:10

**Comment by alanfx** [comment]
@isavin I guess to me this change just redefines the source hotrod and hotrod-static. One is built from the sources, and one is already built.

**Pull request closed without merging**